### PR TITLE
make headers consistent

### PR DESCRIPTION
**Pull request by YingXue**
Make detail page's header consistent along with tooltips/links
Make device id copiable so it's easier for user to grab it and use it in the query
changed the order of pnp pages, so readonly pages and writable pages are grouped logically

### CODE REVIEW

**Comment by YingXue** [comment]
> Thinking renderHeader should be a component -- a quick functional component
> 
> export const Header: React.FC = props => {
> return
> 
> ### t(props.key)
> 
> }
> might better align with the react 'way'

Good idea. I am able to convert this to FC with the exception of not able to use 'useLocalizationContext()' as getting runtime error. Would like to borrow some help tomorrow.

File: src/app/devices/deviceContent/components/deviceContent.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export class DeviceContentComponent extends React.PureComponent<DeviceContentPro
                             <div className="device-content">
                                 <>
                                     {this.renderNav(context)}
-                                    {this.renderDeviceContentDetail()}
+                                {this.renderDeviceContentDetail()}
```
Comment (asrudra) [inline src/app/devices/deviceContent/components/deviceContent.tsx]: nit: indentation

File: src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx [hunk 2]
```diff
@@ -104,8 +106,30 @@ export default class DeviceInterfaces extends React.Component<DeviceInterfacePro
...
                         {context.t(ResourceKeys.deviceInterfaces.command.configure)}
                 </ActionButton>
-                <Label> {context.t(ResourceKeys.deviceInterfaces.columns.displayName)}: {displayName}</Label>
-                <Label> {context.t(ResourceKeys.deviceInterfaces.columns.description)}: {description}</Label>
+                <MaskedCopyableTextFieldContainer
+                    ariaLabel={context.t(ResourceKeys.deviceInterfaces.columns.id)}
+                    label={context.t(ResourceKeys.deviceInterfaces.columns.id)}
+                    value={getInterfaceIdFromQueryString(this.props)}
+                    allowMask={false}
+                    t={context.t}
```
Comment (asrudra) [inline src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx]: why do we need to pass "t" ?

File: src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx [hunk 3]
```diff
@@ -140,7 +164,7 @@ export default class DeviceInterfaces extends React.Component<DeviceInterfacePro
                                 {(themeContext: ThemeContextInterface) => (
                                     <Editor
                                         language="json"
-                                        height="calc(100vh - 400px)"
+                                        height="calc(100vh - 600px)"
```
Comment (asrudra) [inline src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx]: not a fan of inline css

File: src/app/shared/components/headerView.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { useLocalizationContext } from '../contexts/localizationContext';
+
+export interface HeaderViewProps {
+    headerText: string;
+    tooltip?: string;
+    link?: string;
+}
+
+export const HeaderView: React.FC<HeaderViewProps> = props => {
+    const hostId = getId('tooltip');
```
Comment (asrudra) [inline src/app/shared/components/headerView.tsx]: why is this needed?

File: src/app/shared/components/headerView.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const { headerText, tooltip, link } = props;
+
+    const { t } = useLocalizationContext();
+    return (
+        <Stack horizontal={true}>
+            <Stack.Item align="start">
+                <h3> {t(headerText)}</h3>
```
Comment (asrudra) [inline src/app/shared/components/headerView.tsx]: nit: extra space

File: src/app/shared/components/headerView.tsx (hunk 4 above)
Comment (YingXue) [reply src/app/shared/components/headerView.tsx]: for accessibility to asscociate the tooltip to the text

File: src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx (hunk 2 above)
Comment (YingXue) [reply src/app/devices/deviceContent/components/deviceInterfaces/deviceInterfaces.tsx]: wasn't going to change this one. but since the pr is paying off some debt, sure why not.
